src/main/java/com/gregtechceu/gtceu/api/data/worldgen/BiomeWeightModifier.java

```diff
@@ -33,4 +34,19 @@ public BiomeWeightModifier(Supplier<HolderSet<Biome>> biomes, int addedWeight) {
     public Integer apply(Holder<Biome> biome) {
         return biomes.get().contains(biome) ? addedWeight : 0;
     }
+
+    @Override
+    public final boolean equals(Object object) {
+        if (this == object) return true;
+        if (!(object instanceof BiomeWeightModifier that)) return false;
+
+        return addedWeight == that.addedWeight && biomes.get().equals(that.biomes.get());
```

Comment:
Could the HolderSet supplied ever be null?

Reply:
not in there, no.